### PR TITLE
Catalog-of-my-things

### PR DESCRIPTION
**Pull request by OmarMWarraich**
# Worthy Code Reviewer,

<div style="width:100%;height:0;padding-bottom:100%;position:relative;"></img></div>

## In this PR, following requirements have been fulfilled:

Project requirements
Logic

    Each class from the [UML class diagram](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/images/catalog_of_my_things.png) should be created in a separate .rb file.
    All associations (1-to-many relationships) and aggregations (parent-child relationships) between classes should be implemented.
    All properties visible in the diagram should be defined and set up in the constructor method. Exception: properties for the 1-to-many relationships should NOT be set in the constructor method. Instead, they should have a custom setter method created.
    All methods visible in the diagram should be implemented:
        add_item method (in all classes that have that method)
            should take an instance of the Item class as an input.
            should add the input item to the collection of items.
            should add self as a property of the item object (by using the correct setter from the item object).
        can_be_archived?() in the Item class
            should return true if published_date is older than 10 years.
            otherwise, it should return false.
        move_to_archive() in the Item class
            should reuse can_be_archived?() method.
            should change the archived property to true if the result of the can_be_archived?() method is true.
            should do nothing if the result of the can_be_archived?() method is false.
        can_be_archived?() in the Book class
            should override the method from the parent class.
            should return true if parent's method returns true OR if cover_state equals to "bad".
            otherwise, it should return false.
        can_be_archived?() in the MusicAlbum class
            should override the method from the parent class.
            should return true if parent's method returns true AND if on_spotify equals true.
            otherwise, it should return false.
        can_be_archived?() in the Movie class
            should override the method from the parent class.
            should return true if parent's method returns true OR if silent equals true.
            otherwise, it should return false.
        can_be_archived?() in the Game class
            should override the method from the parent class.
            should return true if parent's method returns true AND if last_played_at is older than 2 years.
            otherwise, it should return false.
    Add unit tests for all implemented methods.

User interface

    Create a main.rb file that will serve as your console app entry-point.
    Your console app, at the start, should:
        Present the user with a list of options to perform.
        Let users choose an option.
        If needed, ask for parameters for the option.
        Have a way to quit the app.
    The following options should be available:
        List all books
        List all music albums
        List all movies
        List of games
        List all genres (e.g 'Comedy', 'Thriller')
        List all labels (e.g. 'Gift', 'New')
        List all authors (e.g. 'Stephen King')
        List all sources (e.g. 'From a friend', 'Online shop')
        Add a book
        Add a music album
        Add a movie
        Add a game
    All data should be preserved by saving collections in .json files.

Database

    Create a schema.sql file with tables that will be analogical to the structure of the classes in your app. As you cannot implement inheritance in the database tables - add all properties and associations from the parent Item class as table columns to all tables based on the child classes. Thanks to that it can be used to store data in the future.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,189 @@
...
+## Tests 
+the app is tested using Rspec library you can check the tests by openning your terminal, then change the directory to the location of this repo using this command `cd file_path`, then run this command `gem install rspec`, finally run this command `rspec spec` or `rspec spec --format doc` for documented test results you should see a screen like that 
+
+![test1](https://user-images.githubusercontent.com/57269872/206801328-3a77f3b0-0a03-4488-afae-8c528ce9fe78.jpeg)
+
+![test2](https://user-images.githubusercontent.com/57269872/206801452-64d0724e-5d60-4b97-b4ba-f30b6d76569d.jpeg)
+
+## Video Summary
+
+[Video](https://www.loom.com/share/2430bfc023674005ba1930c14fac834a)
```
Comment (AnselemOdims) [inline README.md:128]: - [ ] Good job recording your presentation with your teammates, but only one member of the presentation is audible. Please record it again, this time every member should be audible. Also, do not forget to highlight an interesting piece of code you are proud of.

File: database/schema.sql [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    FOREIGN KEY(game_id) REFERENCES Game(id),
+    FOREIGN KEY(music_album_id) REFERENCES Music_Album(id)
+);
+
+CREATE TABLE Game (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  publish_date DATE,
+  archived BOOLEAN,
+  multiplayer BOOLEAN,
+  last_palayed_at DATE,
```
Comment (AnselemOdims) [inline database/schema.sql:29]: - [ ] Good job on writing the queries for creating the tables, but please kindly ensure all foreign is indexed.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @team, 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by KayLemba — APPROVED** [review]
[# STATUS: APPROVED :green_circle: :clap: 

Hello  @TEAM     , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

HIGHLIGHTS:

- [x]  Your repo is professional 
- [x] Gitflow is followed
- [x] Descriptive P.R
- [x] Descriptive README file




 

- [x] No Linter Errors







Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]

**Pull request merged**